pkg/kclient/services.go

```diff
@@ -52,7 +53,10 @@ func (c *Client) DeleteService(serviceName string) error {
 // GetOneService retrieves the service with the given component and app name
 // An error is thrown when exactly one service is not found for the selector.
 func (c *Client) GetOneService(componentName, appName string) (*corev1.Service, error) {
-	return c.GetOneServiceFromSelector(componentlabels.GetSelector(componentName, appName))
+	labels := componentlabels.GetLabels(componentName, appName, false)
+	labels[componentlabels.OdoModeLabel] = componentlabels.ComponentDevName
```

Comment:
Are we sure it'll always be "Dev" mode? To me this method seems generic, so it should ask for the mode, instead of assuming it. What do you think?

Reply:
For me, yes it will be always Dev, as we specifically handles Deployments and Services in Dev mode only (In deploy mode, we are managing Kubernetes resources, not specifically Deployments and Services).
These methods should be better placed in some Dev specific package